### PR TITLE
fix(PPDSC-2608): fix select panel position on scroll

### PR DESCRIPTION
**Pull request by mutebg**
PPDSC-2608

**What**

1. Background - when select panel is open and user scroll the content depending on the HTML structure ( where the scroll is ) the panel might not move together with the Selelect Button.
2. What did you do: Attachet event listener to the parent scrollable element, for that I used a helper function from `floating-ui` which find the ancestors with overflow. With that when the user scrolls, new dimensions have been taken and position updated
3. What does the reviewers should expect 

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

File: src/screen-reader-only/screen-reader-only.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ const ScreenReader = styled.span`
   position: absolute;
   white-space: nowrap;
   width: 1px;
+  margin: -1px;
```
Comment (mstuartf) [inline src/screen-reader-only/screen-reader-only.tsx:17]: What's the reason for this change?

**Review by mstuartf — APPROVED** [review]

File: src/screen-reader-only/screen-reader-only.tsx (hunk 1 above)
Comment (mutebg) [reply src/screen-reader-only/screen-reader-only.tsx:17]: by default, the size is width: 1px; height:1px but that adds a horizontal scroll  in some cases, like imagine this screen-reader component being placed together with another element which is 100vw, this is what was happening in Dow scenario, the negative margin removes that sizing

**Review by baburay23 — APPROVED** [review]
